### PR TITLE
Implement static stylesheet extraction

### PR DESCRIPTION
**Pull request by felixjung**
Opening this so you can follow along, if you're interested.

**TODO**
- [x] List all components.
- [x] Extract component meta data for automatic rendering of different props.
- [x] Render each component with every modifier style.
- [x] Extract rendered styles for each component from emotion cache.
- [x] Clean up class names.
- [x] Write static stylesheet to disk.

### CODE REVIEW

**Comment by connor-baer** [comment]
@felixjung Is this ready for review?

**Comment by felixjung** [comment]
Yes

File: build/stylesheet/lib/extract-styles.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import * as emotion from 'react-emotion';
+import * as ReactTestRenderer from 'react-test-renderer';
+import { ThemeProvider } from 'emotion-theming';
+
+import { standard } from '../../../src/themes';
```
Comment (connor-baer) [inline build/stylesheet/lib/extract-styles.js]: Do we want to make the theme configurable?

File: build/stylesheet/lib/parse-component.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    'SIZES.KILO': 'kilo',
+    'SIZES.MEGA': 'mega',
+    'SIZES.GIGA': 'giga',
+    'SIZES.TERA': 'tera',
+    'SIZES.PETA': 'peta',
+    'SIZES.EXA': 'exa',
+    'SIZES.ZETTA': 'zetta',
+    true: true,
+    false: false
+  };
```
Comment (connor-baer) [inline build/stylesheet/lib/parse-component.js]: Do we need to replace all constants?

File: build/stylesheet/lib/extract-styles.js (hunk 1 above)
Comment (felixjung) [reply build/stylesheet/lib/extract-styles.js]: Yes. Makes total sense for when we start polishing this for release.

File: build/stylesheet/lib/parse-component.js (hunk 2 above)
Comment (felixjung) [reply build/stylesheet/lib/parse-component.js]: Yes! Ideally programmatically at some point.

**Pull request merged**